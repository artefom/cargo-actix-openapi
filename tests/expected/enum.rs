//! API auto-generated by apigen

use std::fmt::Display;

use std::{collections::HashMap, fmt::Debug};

use serde::{Deserialize, Serialize};

use actix_web::{
    get, http::StatusCode, middleware::NormalizePath, web, App, HttpRequest, HttpResponse,
    HttpServer, ResponseError,
};

use async_trait::async_trait;

// Enums
// -------------------------------

/// String enum example
#[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq)]
pub enum GreetUserResponseStrEnum {
    #[serde(rename = "First Variant")]
    FirstVariant,
    #[serde(rename = "Second variant $")]
    SecondVariant,
    #[serde(rename = "!123")]
    V123,
    #[serde(rename = "Hello, \"World\"")]
    HelloWorld,
    #[serde(rename = "Hello, \\\"World2\\\"!")]
    HelloWorld2,
}


// Struct
// -------------------------------

#[derive(Serialize, Deserialize, Clone, PartialEq)]
pub struct GreetUserPath {
    pub user: String,
}

/// Enum container
#[derive(Serialize, Deserialize, Clone, PartialEq)]
pub struct GreetUserResponse {
    #[serde(rename="strEnum")]
    pub strenum: GreetUserResponseStrEnum,
    #[serde(rename="intEnum")]
    pub intenum: i64,
}
