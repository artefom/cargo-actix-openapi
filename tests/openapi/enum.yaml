openapi: "3.0.3"
info:
  version: "1.0"
  title: "Hello World API"
paths:
  /hello/{user}:
    parameters:
      - name: user
        in: path
        required: true
        description: The name of the user to greet.
        schema:
          type: string
    get:
      operationId: greet_user
      summary: Returns a greeting to the user!
      description: Some detailed description
      responses:
        200:
          description: Returns the greeting.
          content:
            application/json:
              schema:
                type: object
                description: "Enum container"
                required:
                  - strEnum
                  - intEnum
                properties:
                  strEnum:
                    type: string
                    description: "String enum example"
                    enum:
                      [
                        "First Variant",
                        "Second variant $",
                        "!123",
                        'Hello, "World"',
                        'Hello, \"World2\"!',
                      ]
                  intEnum:
                    type: integer
                    description: "Integer enum example"
                    enum: [1, 2, 3]
