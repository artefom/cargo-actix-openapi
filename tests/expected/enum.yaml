api:
  definitions:
  - name: GreetUserPath
    data: !Struct
      doc: null
      properties:
        user: String
  - name: GreetUserResponseStrEnum
    data: !Enum
      doc: String enum example
      variants:
      - name: FirstVariant
        value: First Variant
      - name: SecondVariant
        value: Second variant $
      - name: V123
        value: '!123'
      - name: HelloWorld
        value: Hello, "World"
      - name: HelloWorld2
        value: Hello, \"World2\"!
  - name: GreetUserResponse
    data: !Struct
      doc: Enum container
      properties:
        strEnum: GreetUserResponseStrEnum
        intEnum: i64
  operations:
  - name: greet_user
    path: /hello/{user}
    method: Get
    doc: Returns a greeting to the user!
    param_path: web::Path<GreetUserPath>
    param_query: null
    param_body: null
    response: web::Json<GreetUserResponse>
